libcontainer/cgroups/fs/memory.go

```diff
@@ -147,6 +141,40 @@ func memoryAssigned(cgroup *configs.Cgroup) bool {
 		cgroup.MemorySwappiness != -1
 }
 
+func getMemoryMappingData(path, name string) (map[string]uint64, error) {
+	data := make(map[string]uint64)
+
+	file, err := os.Open(filepath.Join(path, name))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	defer file.Close()
+
+	sc := bufio.NewScanner(file)
+	for sc.Scan() {
+		t, v, err := getCgroupParamKeyValue(sc.Text())
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s (%q) - %v", name, sc.Text(), err)
+		}
+		data[t] = v
+	}
+
+	return data, nil
+}
+
+func getMemoryStat(path string) (map[string]uint64, error) {
+	// Get stats from memory.stat.
+	return getMemoryMappingData(path, "memory.stat")
+}
+
+func getOomStat(path string) (map[string]uint64, error) {
```

Comment:
This is not really a stat. Why are we overloading the stats API to essentially abstract out cgroups API?